charts/cluster-api-runtime-extensions-nutanix/defaultclusterclasses/nutanix-cluster-class.yaml

```diff
@@ -0,0 +1,305 @@
+      generateExtension: nutanixclusterconfigpatch.cluster-api-runtime-extensions-nutanix
+    name: cluster-config
+  - external:
+      discoverVariablesExtension: nutanixworkerconfigvars.cluster-api-runtime-extensions-nutanix
+      generateExtension: nutanixworkerconfigpatch.cluster-api-runtime-extensions-nutanix
+    name: worker-config
+  workers:
+    machineDeployments:
+    - class: nutanix-quick-start-worker
+      machineHealthCheck:
+        maxUnhealthy: 40%
+        nodeStartupTimeout: 10m
+        unhealthyConditions:
+        - status: "False"
+          timeout: 300s
+          type: Ready
+        - status: Unknown
+          timeout: 300s
+          type: Ready
+        - status: "True"
+          timeout: 300s
+          type: MemoryPressure
+        - status: "True"
+          timeout: 300s
+          type: DiskPressure
+        - status: "True"
+          timeout: 300s
+          type: PIDPressure
+        - status: "True"
+          timeout: 300s
+          type: NetworkUnavailable
+      template:
+        bootstrap:
+          ref:
+            apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+            kind: KubeadmConfigTemplate
+            name: nutanix-quick-start-kcfg-0
+        infrastructure:
+          ref:
+            apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+            kind: NutanixMachineTemplate
+            name: nutanix-quick-start-md-nmt
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlaneTemplate
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: nutanix
+  name: nutanix-quick-start-kcpt
+spec:
+  template:
+    spec:
+      kubeadmConfigSpec:
+        clusterConfiguration:
+          apiServer:
+            certSANs:
+            - localhost
+            - 127.0.0.1
+            - 0.0.0.0
+            extraArgs:
+              cloud-provider: external
+              tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
+          controllerManager:
+            extraArgs:
+              cloud-provider: external
+              tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
+          scheduler:
+            extraArgs:
+              tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
+        files:
+        - content: |
+            apiVersion: v1
```

Comment:
i'd like to see this modified with a patch handler, not like this.

Reply:
https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/392#pullrequestreview-1949576783. it was discussed earlier and its good to make everything variable, it was deemed not urgent. you can file a tracker if you think its a must fix with reason at this time

Reply:
I don't think this needs to be handled by a variable, but we can reuse this code if it's implemented as a patch handler.

Reply:
Can we deploy kube-vip as an addon?

Reply:
I don't think so. I'm not sure how a static pod deployment like kube-vip would get deployed as an addon.

It seems like something we want as a static pod

Reply:
Ah good point. But we could do this as a before create cluster hook that adds this static pod in to kubeadm config template, right? This is a generic pod for other providers too, right?

Reply:
Yeah, we want to reuse this for providers like vsphere and for preprovisioned. We could do it as a BeforeClusterCreate hook or add it in as GeneratePatch